src/Components/Feed/Posts.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect} from 'react'
+import { Post } from './Post.jsx'
+import axios from 'axios';
+import { WithPost } from '../../HOCs/WithPost.jsx';
+import { useDoLike } from '../apis/useDoLike.jsx';
+import { useGetPost } from '../apis/useGetPost.jsx';
+import { useDoUnLike } from '../apis/useDoUnlike.jsx';
+import { PostLoader } from '../Loaders/PostLoader.jsx';
+
+export const PostsComponent = ({ statePost, setStatePost }) => {
+  const [postRes, getPost] = useGetPost();
+  const [res, doLike] = useDoLike()
+  const [unlikeRes, doUnLike] = useDoUnLike()
+  const [currentPostId, setCurrentPostId] = useState()
+  const [errorMessage, setErrorMessage] = useState('');
+  let userId = null
+
+  useEffect(() => {
+    getPost()
+  }, [])
+
+  useEffect(() => {
+    console.log(postRes)
+    if (postRes.data) {
+      setStatePost(postRes.data)
+    }
+  }, [postRes])
+
+  useEffect(() => {
+    const index = statePost?.findIndex(statePost => statePost._id === currentPostId)
+    const tempPosts = [...statePost]
+    if (res.status === 200) {
+      tempPosts?.[index]?.likes.push(res.data)
+      setStatePost(tempPosts)
+      setErrorMessage('')
+    }
+    else if (res.error === 400) {
+      setErrorMessage("You have already Liked")
+      console.log(errorMessage)
+    }
+
+  }, [res, currentPostId])
+
+  useEffect(() => {
+    const index = statePost.findIndex(statePost => statePost._id === currentPostId)
+    const tempPosts = [...statePost]
+    if (unlikeRes.status === 200) {
+      tempPosts[index].likes.pop()
```

Comment:
how can you be sure current user's like will always be the last item of likes array.